### DISCUSSION TITLE
Batch-user creation in ltsp19 with LTSP Manager?

### DISCUSSION THREAD

**Discussion by dgroos** [ltsp/ltsp#193] [Q&A]
I've got an Ubuntu 18.04, LTSP19, 2-NIC system setup and working in my classroom. For simplicity sake, I'm going to create ~140 local user accounts instead of joining the district AD authentication system as I did w/ ltsp5.

I've read about batch user-creation apps eg [newusers](http://manpages.ubuntu.com/manpages/xenial/man8/newusers.8.html) where I create a text file with needed info and it creates the accounts. However, I don't know if this would work with ltsp19 or if it needs specific settings.

I saw that https://github.com/ltsp/community/issues/26#issuecomment-539528498 mentions that LTSP Manager might just work for LTSP19, too. Has anyone tried it for user creation in Ubuntu 18.04?

**Comment by enaut** [comment]
I have not tried I think you will need to use the adjusted ltsp-manager from my ppa repos: see these instructions: http://wiki.ltsp.org/wiki/Ltsp-manager/Installation I will probably look into it in the coming month or so. But so far batch user creation should "just work". Give it a try and if it does not work you can complain here... I will do my best to fix everything.

Also note that in this ppa is a python script with launcher button and all. called "password-changer". This enables every user to change their password on the server (with ssh to server and passwd command but all through a nice and simple gui).

**Comment by dgroos** [comment]
Hi @enaut thanks for starting to move ltsp-manager to the new ltsp19 platform and offering to troubleshoot things as I batch-add. I'm going to set up a virtual testing environment so I can roll back to a snapshot if necessary.

Looking at your linked instructions, I'll first add the Greek schools ppa, then yours since I'm working on Ubuntu 18.04.

My question/concern is that, if I've already installed and setup ltsp19, when I install ltsp-manager (which is set to work w/ltsp5), won't it interfere with epoptes, dnsmasq, etc. in the 2 NIC, ltsp19 setup I'll already have working?

Thanks!
David

**Comment by enaut** [comment]
Hi @dgroos , alkis said that it is no problem to install old and new ltsp in parallel.

you can also try to only install the new ppa!
ltsp-manager does not depend on much so it might be all satisfied without anything from the old ppa.

so actually it's just:
```
root@server # add-apt-repository ppa:franz-die/ltsp-manager; apt update
root@server # apt install ltsp-manager
```

and most parts of ltsp-manager should work.

**Comment by alkisg** [comment]
If I recall correctly, ltsp-manager Recommends: the LTSP5 packages, but it doesn't Depends: on them.

So if one uses `apt install --no-install-recommends ltsp-manager`, they should be able to install LTSP manager without any LTSP5 packages.

Btw, the wiki.ltsp.org wiki will be closed soon; I've created a new one at https://github.com/ltsp/ltsp5/wiki; maybe we can import the content there, so that it's archived.
I'm not sure if LTSP manager belongs there, or if it should go to its own ltsp/ltsp-manager wiki, or if it should go to the new ltsp/ltsp wiki.

**Comment by rkwesk** [comment]
@dgroos 
As someone who has tested ltsp19 with 2 nics I call your attention to how you invoked the dnsmasq conf command. I don't know how or if the ltsp manager deals with the dnsmasq conf file but be aware what file(s) are in /etc/dnsmasq.d/ as the ltsp5 file (ltsp-server-dnsmasq.conf) would conflict with the ltsp19 file (ltsp-dnsmasq.conf) and ltsp19 will halt the configuration of dnsmasq if there is any such "old" file.

In the forum in Greek Alkis has told us about a command that clears up which version has the higher priority and would be considered the "latest version" in the event that a system has ppa repos. It is:

~~~
     apt policy epoptes
     apt policy dnsmasq
~~~

Therefore you can see that the latest ltsp19 packages would prevail in any case, even with various ppa repos added.

Richard

**Comment by dgroos** [comment]
Thanks @enaut and @alkisg I'll get back to you on this, and thanks Richard for the apt policy command. I will be running it and the results will give either troubleshooting info or peace of mind!
David

**Comment by enaut** [comment]
@dgroos I updated my old testing repository with a version that has the initial-setup and server-menu disabled. that way you will probably destroy less ;)

https://launchpad.net/~franz-die/+archive/ubuntu/ltsp-manager-testing

```
# sudo add-apt-repository ppa:franz-die/ltsp-manager-testing
# sudo apt install ltsp-manager
```

I am not using ubuntu on my personal computers and have currently no access to any virtual machine so I havn't been able to test it in ubuntu however in fedora it runs fine.

**Comment by dgroos** [comment]
Thanks for this! Been working under a deadline so haven't been able to
finish testing but expect to be able to do so this weekend and will let you
know how things go.
David

**Comment by dgroos** [comment]
I'm at the point where I want to upload the csv of user-info to ltsp-manager for batch user creation and haven't been able to find info on formatting the csv. Someone have that info? Thanks!

**Comment by alkisg** [comment]
Extract your users into .csv, then use that as a template.

**Comment by dgroos** [comment]
I used your suggestion @alkisg removing the entry for "encrypted password" and adding the password I'm assigning to the (test) users in the Password "column". Seems to work great so far!

2 questions:

1. I've not seen any issues in the VM testing setup so far--that is great! What log/s should I check or what should I do to incite problems during my testing of LTSP-Manager? Trying to find lurking problems here!

1. I'll be having about 150 users, and this is a lot of scrolling for each login. Is there some search-as-you-type capability for LightDM?

Thanks,
David

**Comment by alkisg** [comment]
For (2), LightDM only shows the first 50 users, so the rest will need to type in their username in the "other user" text box.
To customize which users will show up in LightDM on each client (again up to 50), read https://github.com/ltsp/ltsp/issues/33 and comment there if you still have questions.

Edit: but... are you sure you're using LightDM? If I recalled correctly, you were using Ubuntu GNOME, which uses GDM...

**Comment by enaut** [comment]
In GDM it is most practical to just disable the userlist completely. So far LTSP does not have the option but you can easily do so with dconf:
https://help.gnome.org/admin/system-admin-guide/stable/login-userlist-disable.html.en

With the user creation not much has changed so the creation should work without issues (if something is created at all).

That aside: are you still using the testing ppa? If so be careful with updates... as I'm testing some more.

**Comment by dgroos** [comment]
You're right @alkisg I am using GDM (I didn't really differentiate between them). I thought that ltsp19 used LightDM as a standard. From the bit of reading I've done, sounds like LightDM might be the better option. [I see this page telling how to change display managers](https://www.linuxuprising.com/2018/12/how-to-change-default-display-manager.html). Is it recommended that I switch?

**Comment by dgroos** [comment]
@enaut I was going to install/use ltsp-manager (from your testing ppa) this afternoon (8-9 hours from now). How should I proceed? Thanks!

**Comment by enaut** [comment]
@dgroos use the "normal" one - I updated it so that it is the version that has previously been in testing:
```
# sudo add-apt-repository ppa:franz-die/ltsp-manager
# sudo apt install ltsp-manager
```
regarding gdm vs lightdm - actually it does not matter anymore as ltsp authenticates with pam nowadays Its just a matter of preferences.

**Comment by dgroos** [comment]
@enaut OK thanks will update (and assume I just go into sources and comment out the lines for your testing one).

About DM's, that's good to know! I'll be checking ltsp/ltsp#33 and dconf links given above.

**Comment by alkisg** [comment]
> Is it recommended that I switch?

Nah. The philosophy is: you find out how to manually configure your DM the way you want it (by reading/googling about GDM configuration), and then you file an ltsp issue and ask developers to make it easier via ltsp.conf parameters.

Only if the DM can't do what you want, it then makes sense to switch to another...
